src/esm_parser/esm_parser.py

```diff
@@ -1527,6 +1527,85 @@ def resolve_choose(model_with_choose, choose_key, config):
         logging.debug("key=%s", key)
 
 
+def resolve_early_var(
+    var, config, current_model=None, user_config={}, model_config={}, setup_config={}
+):
+
+    #import ipdb
+    if not current_model:
+        logger.error(
+            "Please define a ``current_model`` for ``resolve_early_var`` method"
+        )
+        sys.exit(1)
+    elif current_model in config:
+        this_config = config
+    elif current_model == config.get("model"):
+        this_config = {current_model: config}
+    else:
+        logger.error(f"``current_model: {current_model}`` and ``config`` do not correspond!")
+        return
+
+    resolve_choose_with_var(
+        var,
+        this_config,
+        current_model=current_model,
+        user_config=user_config,
+        model_config=model_config,
+        setup_config=setup_config,
+    )
+
+    if var not in this_config[current_model]:
+        logger.debug(f"``{var}`` not found in this config (resolve_early_var)")
+        return
+
+    value = this_config[current_model].get(var)
+
+    #if value==['xios']:
+        #ipdb.set_trace()
+
+    if isinstance(value, str) and "${" in value:
+        value_pattern = r'\${(.*?)}'
+        subvars = re.findall(value_pattern, value)
+        for subvar in subvars:
+            path_to_var = subvar.split(".")
+            if len(path_to_var) > 1:
+                subvar_component = path_to_var[0]
+                subvar = ",".join(path_to_var[1:])
+            else:
+                subvar_component = current_model
+            #if var == "version" and var in config[current_model] and current_model=="fesom":
+                #ipdb.set_trace()
+            resolve_early_var(subvar, config, current_model=subvar_component, user_config=user_config, model_config=model_config, setup_config=setup_config)
+
+            config_to_search_into = None
+            if subvar in user_config.get(subvar_component, []):
+                config_to_search_into = user_config
+            elif subvar in setup_config.get(subvar_component, []):
+                config_to_search_into = setup_config
+            elif subvar in model_config.get(subvar_component, []):
+                config_to_search_into = model_config
+            else:
+                return
+
+            try:
+                new_value = find_variable([current_model], this_config[current_model][var], config_to_search_into, [], True)
+            except esm_parser.esm_parser.EsmParserError as error:
+                not_found_variable = str(error).split("was not resolved: ")[1].split(" not found")[0]
```

Comment:
Interesting idea:

Can `Error` objects return things or have attributes?

Yes. See this example:
```python
class EsmParserError(Exception):
    """Base class for exceptions in the ESM parser."""
    def __init__(self, message="An error occurred in the ESM parser"):
        self.message = message
        super().__init__(self.message)

class EsmParserVariableNotFoundError(EsmParserError):
    """Exception raised when a variable is not found in the ESM parser."""
    def __init__(self, variable_name):
        self.variable_name = variable_name
        self.message = f"Variable '{self.variable_name}' not found in the ESM parser."
        super().__init__(self.message)

# Example usage
try:
    # Simulate code that may raise EsmParserVariableNotFoundError
    raise EsmParserVariableNotFoundError("my_variable")
except EsmParserVariableNotFoundError as e:
    # Extract the variable name from the exception and perform some logic with it
    variable_name = e.variable_name
    capitalized_variable_name = variable_name.capitalize()
    print(f"Original variable name: {variable_name}")
    print(f"Capitalized variable name: {capitalized_variable_name}")
```

This might be worth using in a future implementation...